src/core/struct_tree.js

```diff
@@ -388,11 +394,12 @@ class StructTreeRoot {
     }
 
     const id = pageDict.get("StructParents");
-    const numberTree = new NumberTree(parentTree, xref);
-    const parentArray = numberTree.get(id);
-    if (!Array.isArray(parentArray)) {
+    if (!Number.isInteger(id)) {
       return;
     }
+    const numberTree = new NumberTree(parentTree, xref);
+    const parentArray = numberTree.get(id);
```

Comment:
Don't you still need to check that `parentArray` is actually an array here, as was done previously, to prevent any errors below?

Reply:
Normally, it has already been done there:
https://github.com/mozilla/pdf.js/pull/17043/files#diff-1bcdf33154f754078d7eff4ac8b496e5fc7b2813faaee0f264fc9e0667aee132R194
isn't enough ? or could you imagine something where my assertion is wrong ?

Reply:
Assuming that we're not calling `#collectParents` from somewhere else, I'd agree with your analysis here.